src/main/java/frc/robot/subsystems/shooter/commands/Fire.java

```diff
@@ -0,0 +1,89 @@
+package frc.robot.subsystems.shooter.commands;
+
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpiutil.math.MathUtil;
+import frc.robot.Robot;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.shooter.Shooter;
+
+import static frc.robot.Constants.Shooter.*;
+
+/**
+ * This class fires the shooter
+ * sniper - this variable is the shooter
+ * distance - this is the target distance
+ * targetSpeed - this is the speed the motors
+ * need to get to
+ */
+public class Fire extends CommandBase {
+    private Shooter sniper;
+    private double targetSpeed;
+
+    /**
+     * Constructor
+     *
+     * @param sniper is the Shooter
+     */
+    public Fire(Shooter sniper) {
+        this.sniper = sniper;
+        addRequirements(sniper);
+    }
+
+    /**
+     * This function simply gets the value
+     * for the target speed
+     */
+    @Override
+    public void initialize() {
+        targetSpeed = SPEED;
+    }
+
+    /**
+     * This function runs in a loop and sets
+     * the velocity required according to the
+     * Shooter class function - setVelocity
+     */
+    @Override
+    public void execute() {
+        sniper.setVelocity(targetSpeed);

```

Comment:
calculate the velocity based on the distance

Reply:
No